### ISSUE TITLE
Tentacle poll-server only works for deployment targets in HA

### ISSUE THREAD

**Issue by MichaelJCompton** [OctopusDeploy/Issues#5206]
# The bug

When registering a polling worker in an HA setup, 

`Tentacle.exe register-worker ...server 1...`

followed by

`Tentacle.exe poll-server ...server 2...` 

doesn't work correctly.  It checks if the tentacle is registered as a deployment target and fails.

## What I expected to happen

It should be possible to register a polling worker against multiple Octopus servers in HA.

**Comment by StephenShamakian** [comment]
Thanks for looking into this! This is a blocker for us to productionize any workers with our HA cluster.

**Comment by StephenShamakian** [comment]
@MichaelJCompton any traction on this? I can help debug the issue if you wish or if you have a alpha version of the tentacle installer I can give it a shot see if it fixed the issue.

Coping Derek for reference: @DerekCampbellOctopus

**Comment by DerekCampbellOctopus** [comment]
Additional report (private link) https://secure.helpscout.net/conversation/826083130/43406?folderId=557077

**Comment by hcornelison** [comment]
I also noticed this issue (I believe) as soon as I upgraded from 2018.x.x LTS to 2019.3.1 LTS and started using the new v4 Tentacle Installer.

**TLDR: I was able to just using the v3.24.0 MSI for the Tentacle to get my HA configuration to work properly. Octopus Server will upgrade the Tentacles, and it still works. If I use the v4 Tentacle MSI to install a tentacle, HA mode is busted.**

In my case, I already had a HA cluster up and running, two nodes (NODE1 and NODE2), and each polling Tentacle was installed the same way, I ran the tentacle installer, then from the command line did the commands needed to configure it, in which case I ran the register-with command against NODE1, and then poll-server command ran against NODE2. This worked fine using the old 3.24.0 Tentacle MSI and 2018.x.x LTS.

I upgraded to 2019.3.1 LTS, and installed a couple of new Tentacles with the v4 MSI, same commands, and the only server that the Tentacle would talk to was the one I ran the register-with command against. The server I specified in the poll-server command would show a client connection, but that the certificate thumbprint from the client was not trusted.

I tried running register-with against both server nodes from the tentacle, this actually cleared the certificate thumbprint issue on the server; however, the tentacles would still not work in a HA mode for me. Here was that scenario:

I ran register-with against NODE1 and NODE2 (in that order).
I shut down NODE2.
The Tentacle Log shows that it just kept looping over and over again failing to connect to NODE2.
Deployment was queued from NODE1.
The Task would just timeout, the tentacle would never pick it up.
I turned NODE2 back on, shut NODE1 down.
Everything worked fine from that Tentacle.

Some of the details may be slightly out of order, I burned 6 hours fighting this thinking something else was going on and was getting frustrated. In the end, I used the 3.24.0 MSI to install the tentacle, it worked perfectly fine. Octopus Server upgraded the tentacle, still worked. I could shut down each node, and HA functioned as I would expect. If I followed the exact same steps but with the v4 installer, it was busted. I don't know what could be happening.

**Comment by StephenShamakian** [comment]
@hcornelison I don't believe this is the same issue... 

You mentioned multiple times the "register-with" command.

This issue is with trying to register a worker tentacle using the "register-worker" command in conjunction with the "poll-server" command.

It seems like the "poll-server" command *only* works with the "register-with" command and not the new'ish "register-worker" command for worker nodes. So we are unable to setup worker nodes in a HA cluster.

Also this issue is present in the older versions of the tentacle client installer as well as that's the version I first noticed these issues. I haven't tried "register-worker + poll-server" in the v4 installer yet...

**Comment by matt-richardson** [comment]
I've just attempted to reproduce it, and I think it works fine with version 4.
I setup a 2 node ha cluster, then ran the following commands:

```
.\Tentacle.exe create-instance --instance "MyWorker" --config "C:\Octopus\MyWorker.config"
.\Tentacle.exe new-certificate --instance "MyWorker" --if-blank
.\Tentacle.exe configure --instance "MyWorker" --reset-trust
.\Tentacle.exe configure --instance "MyWorker" --app "C:\Octopus\Applications\MyWorker" --port "10933" --noListen "True"
.\Tentacle.exe polling-proxy --instance "MyWorker" --proxyEnable "False" --proxyUsername "" --proxyPassword "" --proxyHost "" --proxyPort ""
.\Tentacle.exe register-worker --instance "MyWorker" --server "https://matts2nodeha1.example.com" --name "MyWorker" --comms-style TentacleActive --server-comms-port "10943" --username admin --password "SuperSecr3tPassw0rd!" --workerpool "Dev" --force --console
.\Tentacle.exe poll-server --instance "MyWorker" --server "https://matts2nodeha2.example.com" --username admin --password "SuperSecr3tPassw0rd!" --server-comms-port "10943" --console
.\Tentacle.exe service --instance "MyWorker" --install --stop -start
```

and the commands completed successfully. The `Tentacle.Communication.TrustedOctopusServers` setting in the config file successfully listed both servers.

I suspect the issue can be closed.

I did hit an issue where the second node didn't trust the worker until it was restarted, which sounds like your issue, @hcornelison. I will investigate futher, and if necessary, open a new issue.

**Comment by matt-richardson** [comment]
@hcornelison - I've raised https://github.com/OctopusDeploy/Issues/issues/5545 for your issue.

**Comment by StephenShamakian** [comment]
@matt-richardson @DerekCampbellOctopus There seems to still be an issue with this. THe initial error when registering the worker node is gone now in v4.0.1 of the agent. But now Octopus times out when doing a healthcheck after setting the node to draining that the worker was registered with (so only the polling server node is active). I submitted a support email with a image and the log of the healthcheck.

**Comment by matt-richardson** [comment]
I believe this issue was fixed when Tentacle was upgraded to support Spaces - I'm going to close it.

From what I can see, @hcornelison and @StephenShamakian, you're both hitting #5545, which I just merged a fix for, so we should be good.

If you're still facing issues, please let me know.

**Comment by StephenShamakian** [comment]
@matt-richardson @DerekCampbellOctopus I'm assuming this fix was server side? If so was it backported to the 2018 LTS branch? We're running on that version of Octopus at our organization.

**Comment by matt-richardson** [comment]
@StephenShamakian - we are intending on backporting it, but we want to give it a bit of time to bed in on the fast lane before we do that. We're pretty confident in it, but want to make sure that we keep LTS as stable as possible.
